### ISSUE TITLE
Documentation on the classes corresponding to major XML elements?

### ISSUE THREAD

**Issue by ianfixes** [kobsy/juniter#3]
I'm trying to make sure that I'm properly understanding the example in the README:

```ruby
file = Juniter.parse(xml)
failures = file.test_suites.test_suites.first.test_cases.select(&:fail?)
```

I was surprised to see `file.test_suites.test_suites` duplicated -- I thought it was a typo at first.  But what is the difference between `file.test_suites` and `file.test_suites.test_suites`?  What's the proper way to iterate over all test suites & all test cases?

I was looking at https://github.com/kobsy/juniter/blob/master/lib/juniter/test_suites.rb but didn't see much in the way of explanation.  Same for the unit tests.

**Comment by kobsy** [comment]
I agree that the API there is a bit confusing. 😞 There needed to be a way to represent the `<TestSuites/>` tag itself and access its associated properties. But then how do you access its `<TestSuite/>` element children? Although less than ideal, for the initial implementation I ended up going with the straightforward solution of having a `TestSuites` element object which itself has a `test_suites` method which returns an array of its `TestSuite` children.

I think a future version of this could potentially make the `TestSuites` class include `Enumerable`, but then delegate all the enumerable methods to its now-private `test_suites` array. That could potentially be the best of both worlds -- not needing to seemingly duplicate the method call while still being able to access the various properties of the element and its children -- though I haven't examined what edge cases there might still be lurking in that approach. 🤔

**Comment by ianfixes** [comment]
That more or less describes what I had expected to find -- some sort of `file.each_testcase` enumerable, possibly relying on a `file.each_testsuite` as part of its implementation.
